agent/api/task.go

```diff
@@ -76,6 +77,8 @@ type Task struct {
 	Version string
 	// Containers are the containers for the task
 	Containers []*Container
+	// Resources are the resources for the task
+	Resources []taskresource.TaskResource
```

Comment:
Shouldn't this be a pointer/reference?

Reply:
i think its an interface which are always a pointers

Reply:
`TaskResource` is an interface, hence not adding reference.